dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteBrowserHelper.java

```diff
@@ -177,5 +183,59 @@ public String getSelectedSite(final List<Map<String, Object>> siteList, final St
 		}
 		return selectedSite;
 	}
+	
+	/**
+	 * Returns a subset of the list of sites that the given user has access to
+	 * by query, page and limit 
+	 * .
+	 *
+	 * @param showArchived
+	 *            - Is set to {@code true}, archived sites will be returned.
+	 *            Otherwise, set to {@code false}.
+	 * @param user
+	 *            - The {@link User} performing this action.
+	 * @param filter
+	 *            - (Optional) If specified, returns the sites whose name starts
+	 *            with the value of the {@code filter} variable.
+	 *            
+	 * @param currentPage
+	 *           - indicate the page to obtain the subset of sites to be returned.
+	 *           
+	 * @param sitesPerPage
+	 *           - indicates how many site should be included in the subset to be returned.
+	 *            
+	 * @param respectFrontendRoles
+	 *            -
+	 * @return The list of sites that the given user has permissions to access.
+	 * @throws DotDataException
+	 *             An error occurred when retrieving the sites' data.
+	 * @throws DotSecurityException
+	 *             A system error occurred.
+	 */
+    public Map<String,Object> getPaginatedOrderedSites(final boolean showArchived, final User user, final String filter, final int currentPage, final int sitesPerPage, final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+    	final String sanitizedFilter = filter != null && !filter.equals("all") ? filter : StringUtils.EMPTY;
```

Comment:
filter != null && !filter.equals("all") can be replaced by:
"all".equals (filter)